### PR TITLE
Set passive scan data to scan rules

### PR DESCRIPTION
**Pull request by thc202**
Set the passive scan data to the scan rules and provide the message being scanned through it.
Simplifies the setup both in the production code and tests.
Remove helper test class no longer needed.

### CODE REVIEW

**Review by kingthorin — APPROVED** [review]

**Comment by kingthorin** [comment]
Thanks.

File: zap/src/main/java/org/zaproxy/zap/extension/pscan/PassiveScanTask.java [hunk 1]
```diff
@@ -119,13 +119,11 @@ public void run() {
...
-                            PluginPassiveScanner pps2 = pps.copy();
-                            pps2.init(psThread, msg, passiveScanData);
-                            scanner = pps2;
-                        } else {
-                            scanner.setParent(psThread);
+                            PluginPassiveScanner pps = ((PluginPassiveScanner) scanner).copy();
+                            pps.setHelper(passiveScanData);
+                            scanner = pps;
                         }
+                        scanner.setParent(psThread);
```
Comment (ricekot) [inline zap/src/main/java/org/zaproxy/zap/extension/pscan/PassiveScanTask.java:126]: Do we always want to do this? It wasn't being set before if scanner was not an instance of `PluginPassiveScanner`.

Comment (ricekot) [reply zap/src/main/java/org/zaproxy/zap/extension/pscan/PassiveScanTask.java:126]: Disregard, just looked at `init`.

**Review by ricekot — APPROVED** [review]
Thanks!

**Review by psiinon — APPROVED** [review]

**Pull request merged**